Source/WebCore/Modules/url-pattern/URLPatternParser.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+namespace WebCore {
+namespace URLPatternUtilities {
+
+struct Token;
+enum class TokenType : uint8_t;
+
+enum class PartType : uint8_t { FixedText, Regexp, SegmentWildcard, FullWildcard };
+enum class Modifier : uint8_t { None, Optional, ZeroOrMore, OneOrMore };
+
+struct Part {
+    PartType type;
+    String value;
+    Modifier modifier;
+    String name { };
+    String prefix { };
+    String suffix { };
```

Comment:
What Sihui is saying is not necessarily true for POD structs. It depends on how the struct is being initialized. If you need `{ }` because your local compiler or one of the EWS tells you, this is fine.

Reply:
I'll be clearer:
```
m_partList.append(Part { .type = PartType::FixedText, .value = encodedValue.releaseReturnValue(), .modifier = Modifier::None });
```
This will not build on all compilers WITHOUT the *explicit* initializers that Jessica added. It may build on Mac but I'm fairly confident it won't build on some of the EWS.

Reply:
ah I didn't know we have tried removing it before and ews bots failed. My point is the same: we should add it only when necessary. It seems wpe and gtk bots needed it based on previous test runs, then it's fine to keep it.